src/main/python/systemml/defmatrix.py

```diff
@@ -60,24 +67,42 @@ def __init__(self, inputs, dml=None):
         self.inputs = inputs
         self.dml = dml
         self.ID = None
+        self.depth = 1
         for m in self.inputs:
             m.referenced = m.referenced + [ self ]
+            if isinstance(m, matrix) and m.op is not None:
+                self.depth = max(self.depth, m.op.depth + 1)
 
+    MAX_DEPTH = 0
+    
     def _visit(self, execute=True):
         matrix.dml = matrix.dml + self.dml
 
-    # Don't use this method instead use matrix's printAST()
-    def printAST(self, numSpaces):
+    # Don't use this method instead use matrix's _print_ast()
+    def _print_ast(self, numSpaces):
```

Comment:
Okay the name change for this function `def _print_ast(self, numSpaces)` is good since it shouldn't be called by the user.  However, the other function that is supposed to be used instead should still be called `print_ast()` without an underscore.  Otherwise, they both seem like internal functions.